src/modules/users/users.loader.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@nestjs/common'
+import * as DataLoader from 'dataloader'
+import { NestDataLoader } from 'nestjs-dataloader-dan'
+import { of } from 'rxjs'
+import { concatMap } from 'rxjs/operators'
+import { dlSort } from '../../utils/dlSort'
+import { User } from './models/user.model'
+import { UsersService } from './users.service'
+
+@Injectable()
+export class UsersLoader implements NestDataLoader<string, User> {
```

Comment:
Currently use dataloader to load the user info for each message. In the future use population to decrease the payload size.